### ISSUE TITLE
Deprecate entity_extract_ids() now that we have proper classed entity objects

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#1646]
From https://www.drupal.org/node/1374030. I believe we have "classed entity objects" now right?

**Comment by quicksketch** [comment]
Yep, you can just call `$entity->entityType()`, `$entity->bundle()`, and `$entity->id()` these days. So that function serves no good purpose.

We can't remove functions from 1.x though, but we can mark it deprecated in the docblock indicating it will be removed in the future as well as documenting the alternatives, and we shouldn't be calling it anywhere in core.

**Comment by mikemccaffrey** [comment]
These days, I'm actually leaning in the direction of leaving the layer of these generalized wrapper functions in place (and perhaps even adding a few new ones, as in the case of #862). Even though functional programming has fallen out of vogue, that is what made drupal an easy platform for beginning to intermediate programmers to use, and anything we can do to maintain the ability to interact with the system outside of the non-OOP paradigm would be helpful for continuity.

That being said, entity_extract_ids is a really weird function. If you already know the entity type, you probably also know what the property names are for things like the bundle and id, and can just access them like ```$node->nid```.

Am I correct in my assumptions?

**Comment by gifad** [comment]
@mikemccaffrey : the use case of entity_extract_ids() to write code independant of entity type...
This may be called abstraction, which is devil in backdrop "philosophy"
about philosophy, I've read at https://backdropcms.org/philosophy
> Principals
    Backwards compatibility is important.
    Backdrop will attempt to keep API change to a minimum in order for contributed code to be maintained easily, …

Now the real question is : do you want to make it easy to port modules from drupal 7 to backdrop, or from backdrop to drupal 8 ?
From a programmer's point of view, the answer is not obvious…
